### PR TITLE
swev-id: pylint-dev__pylint-6903 Clamp auto jobs to >=1 to avoid Pool(processes=0) under cgroups (K8s)

### PR DESCRIPTION
**Pull request by rowan-stein**
## Summary
Ensure the auto-detected jobs count from cgroup estimates is never below 1, preventing `multiprocessing.Pool(processes=0)` crashes when running `pylint --jobs=0`.

- Minimal change in `pylint/lint/run.py` `_cpu_count()` to clamp computed value to at least 1.
- Preserves existing logic for cgroups v1/v2 and non-cgroup platforms.

Linked Issue: https://github.com/agyn-sandbox/pylint/issues/32

## Background
In certain Kubernetes/cgroups v1 environments, the newly introduced `_query_cpu()` may yield `0` due to integer truncation:
- `/sys/fs/cgroup/cpu/cpu.cfs_quota_us = -1`
- `/sys/fs/cgroup/cpu/cpu.cfs_period_us = 100000`
- `/sys/fs/cgroup/cpu/cpu.shares = 2` -> `int(2/1024) == 0`

With `--jobs=0`, `_cpu_count()` can then return `0`, leading to:
```text
ValueError: Number of processes must be at least 1
  File "/usr/local/lib/python3.9/dist-packages/pylint/lint/run.py", line 197, in __init__
    linter.check(args)
  File "/usr/local/lib/python3.9/dist-packages/pylint/lint/pylinter.py", line 650, in check
    check_parallel(
  File "/usr/local/lib/python3.9/dist-packages/pylint/lint/parallel.py", line 140, in check_parallel
    with multiprocessing.Pool(
  File "/usr/lib/python3.9/multiprocessing/context.py", line 119, in Pool
    return Pool(processes, initializer, initargs, maxtasksperchild,
  File "/usr/lib/python3.9/multiprocessing/pool.py", line 205, in __init__
    raise ValueError("Number of processes must be at least 1")
```

## Change
In `_cpu_count()`:
- If `cpu_share` is not `None`, return `max(1, min(cpu_share, cpu_count))`.
- Else, return `max(1, cpu_count)`.

This ensures `--jobs=0` never results in `0` workers, and single-process execution is used when cgroup-derived CPU allocation is below 1.

## Reproduction steps (pre-fix)
1) K8s/cgroups v1 environment with values above; run:
```bash
pylint --jobs=0 --verbose my_package
```
Expected: `ValueError: Number of processes must be at least 1` (see stack above).

2) Local monkeypatch:
- Patch `pathlib.Path.is_file` and `open()` to return the values above for cgroups v1 files.
- Run `pylint --jobs=0` on a minimal input.
Expected (pre-fix): same `ValueError`.

## Expected behavior (post-fix)
- `--jobs=0` clamps to `>=1`. When cgroup estimate < 1, jobs == 1; otherwise jobs == min(cgroup_estimate, cpu_count).

## Notes
- We are not running CI per request; unit tests suggested in the Issue for future validation.
- Prior art mentioned by maintainers: pylint PR #6098 and Python bpo-36054.

### CODE REVIEW

File: pylint/lint/run.py [hunk 1]
```diff
@@ -76,8 +76,8 @@ def _cpu_count() -> int:
     else:
         cpu_count = 1
     if cpu_share is not None:
-        return min(cpu_share, cpu_count)
-    return cpu_count
+        return max(1, min(cpu_share, cpu_count))
+    return max(1, cpu_count)
```
Comment (noa-lucent) [inline pylint/lint/run.py:80]: [major] Thanks for tightening the invariant here. Can we add a regression test that proves  never returns 0 when  truncates to zero? This bug slipped in once already, and without a test it can easily regress. One option would be to monkeypatch  to return 0, disable , provide a stub  that returns >1, and assert that  comes back as 1. That would lock this fix in place.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Thanks for addressing the zero-worker crash. The functional change looks good, but we need a regression test locking it in. Please cover the  -> 0 scenario so we don’t lose this fix in a future refactor.

File: pylint/lint/run.py (hunk 1 above)
Comment (rowan-stein) [reply pylint/lint/run.py:80]: Added a regression test at tests/test_cpu_count_minimum.py that monkeypatches _query_cpu() to return 0 and asserts _cpu_count() >= 1. Commit: 0c1cd4741.

**Comment by rowan-stein** [comment]
Addressed review: added regression test to ensure _cpu_count() never returns 0 when _query_cpu() yields 0. Not running CI as requested here, but test is included for future validation.

File: pylint/lint/run.py (hunk 1 above)
Comment (noa-lucent) [reply pylint/lint/run.py:80]: Looks great—thanks for adding the regression coverage.

**Review by noa-lucent — APPROVED** [review]
Regression test looks solid—thanks for closing the loop.
